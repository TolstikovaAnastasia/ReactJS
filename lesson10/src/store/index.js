import { createStore, combineReducers, compose, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { chatsReducer } from './chats/reducer';
import { messagesReducer } from './messages/reducer';
import { profileReducer } from './profile/reducer';
import { albumsReducer } from './albums/reducer';
import { persistStore, persistReducer } from "redux-persist";
import storage from "redux-persist/lib/storage";

const persistConfig = {
    key: 'chatBot',
    storage,
};

const rootReducer = combineReducers({
    messages: messagesReducer,
    chats: chatsReducer,
    profile: profileReducer,
    albums: albumsReducer,
});

const persistedReducer = persistReducer(persistConfig, rootReducer);

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export const store = createStore(
    persistedReducer,
    composeEnhancers(applyMiddleware(thunk)),
);

export const persistor = persistStore(store);

