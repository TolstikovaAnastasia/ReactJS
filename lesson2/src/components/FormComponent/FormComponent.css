.sendInput {
    border: 1px solid black;
    border-radius: 2px;
    padding: 5px;
}

.sendBtn {
    padding: 5px 15px;
}