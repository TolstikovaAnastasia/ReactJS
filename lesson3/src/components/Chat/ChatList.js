import React from 'react';
import { List } from '@mui/material';
import { ListItem } from '@mui/material';

export const ChatList = () => {

    const chatList = [
        {
            title: 'news',
            id: Date.now()
        },
        {
            title: 'music',
            id: Date.now()
        },
        {
            title: 'movies',
            id: Date.now()
        },
    ];

    return (
        <>
            <List>
                {chatList.map(({title}) => (
                    <ListItem key={Date.now()}>
                        {title}
                    </ListItem>
                ))}
            </List>
        </>
    )
}