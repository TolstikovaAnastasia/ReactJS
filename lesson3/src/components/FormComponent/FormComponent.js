import React, {useEffect, useRef, useState} from 'react';
import {Button, TextField} from '@mui/material';

export const FormComponent = ({onAddMessage}) => {
    const [value, setValue] = useState('');
    const inputRef = useRef();

    const handleChange = (e) => {
        setValue(e.target.value);
    };

    const handleSubmit = (e) => {
        e.preventDefault();
        setValue('');
        onAddMessage({text: value, author: 'you', id: Date.now()});
    };

    useEffect(() => {
        inputRef.current?.focus();
    }, []);

    return(
        <>
            <form onSubmit={handleSubmit}>
                <TextField ref={inputRef} autoFocus={true} placeholder="Type message to send" size="small" value={value} onChange={handleChange} />
                <Button type='submit' variant="contained" size="large">Send</Button>
            </form>
        </>
    )
}