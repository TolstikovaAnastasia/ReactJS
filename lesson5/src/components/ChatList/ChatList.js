import React from 'react';
import { Link, Outlet } from 'react-router-dom';
import { List } from '@mui/material';
import { ListItem } from '@mui/material';

export const ChatList = ({chats}) => {
    return (
        <>
            <List>
                {chats.map((chat) => (
                    <ListItem key={chat.id}>
                        <Link to={`/chats/${chat.id}`}>{chat.name}</Link>
                    </ListItem>
                ))}
            </List>
            <Outlet />
        </>
    )
}