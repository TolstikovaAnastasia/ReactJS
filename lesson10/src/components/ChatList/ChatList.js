import React, { useEffect } from 'react';
import { Outlet } from 'react-router-dom';
import { List } from '@mui/material';
import { useDispatch, useSelector } from 'react-redux';
import { FormComponent } from '../FormComponent/FormComponent';
import { selectChats } from '../../store/chats/selectors';
import { ChatItem } from './ChatItem/ChatItem';
import { addChatsWithFb, deleteChatWithFb, initChatsTracking } from '../../store/chats/action';
import { initMsgsTracking } from '../../store/messages/action';

export const ChatList = () => {
    const chats = useSelector(selectChats);
    const dispatch = useDispatch();

    useEffect(() => {
       dispatch(initChatsTracking());
       dispatch(initMsgsTracking());
    }, []);

    const onAddChat = (newChatName) => {
        addChatsWithFb(newChatName);
    };

    const handleDeleteChat = (id) => {
        dispatch(deleteChatWithFb(id));
    };

    return (
        <>
            <List>
                {chats.map((chat) => (
                    <ChatItem key={chat.id} chat={chat} onDelete={handleDeleteChat} />
                ))}
                <FormComponent onSubmit={onAddChat} />
            </List>
            <Outlet />
        </>
    );
};