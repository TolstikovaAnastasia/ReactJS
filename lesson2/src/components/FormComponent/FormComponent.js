import React, {useState} from 'react';
import './FormComponent.css';

export const FormComponent = ({onAddMessage}) => {
    const [value, setValue] = useState('');

    const handleChange = (e) => {
        setValue(e.target.value);
    };

    const handleSubmit = (e) => {
        e.preventDefault();
        setValue('');
        onAddMessage({text: value, author: 'you'});
    };

    return(
        <>
            <form onSubmit={handleSubmit}>
                <input type='text' className='sendInput' placeholder='Type message to send' value={value} onChange={handleChange} />
                <button type='submit' className='sendBtn'>
                    Send
                </button>
            </form>
        </>
    )
}