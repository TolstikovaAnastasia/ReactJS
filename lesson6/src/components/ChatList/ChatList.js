import React from 'react';
import { Link, Outlet } from 'react-router-dom';
import { List } from '@mui/material';
import { ListItem } from '@mui/material';
import { useDispatch, useSelector } from 'react-redux';
import { FormComponent } from '../FormComponent/FormComponent';
import { addChat } from '../../store/chats/action';
import { selectChats } from '../../store/chats/selectors';

export const ChatList = () => {
    const chats = useSelector(selectChats);
    const dispatch = useDispatch();

    const onAddChat = (newChatName) => {
        const newId = `chats${Date.now()}`;
        const newChat = {
            id: newId,
            name: newChatName,
        };
        dispatch(addChat(newChat));
    };

    return (
        <>
            <List>
                {chats.map((chat) => (
                    <ListItem key={chat.id}>
                        <Link style={{textDecoration: 'none'}} to={`/chats/${chat.id}`}>{chat.name}</Link>
                    </ListItem>
                ))}
                <FormComponent onSubmit={onAddChat} />
            </List>
            <Outlet />
        </>
    )
}