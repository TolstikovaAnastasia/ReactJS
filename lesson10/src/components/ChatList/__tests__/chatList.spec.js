import { ChatItem } from '../ChatItem/ChatItem';
import { Provider } from 'react-redux';
import { store } from "../../../store/index";
import { ChatList } from '../ChatList';
import { FormComponent } from '../../../components/FormComponent/FormComponent';
import { render, act, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

describe('ChatList', () => {
    it('', () => {
        const handleDeleteChat = jest.fn();
        const onAddChat = jest.fn();

        const component = render(
            <Provider store={store}>
                <ChatList>
                    <ChatItem key='chat.id' chat='name' text='test name' onDelete={handleDeleteChat} />
                    <FormComponent onSubmit={onAddChat} />
                </ChatList>
            </Provider>
        );

        const submitBtn = component.getByText('Send');

        act(() =>{
            fireEvent.click(submitBtn)
        });

        expect(submitBtn).toBeDefined();
    });

    it("matches snapshot", () => {
        const handleDeleteChat = jest.fn();
        const onAddChat = jest.fn();

        const component = render(
            <Provider store={store}>
                <ChatList>
                    <ChatItem key='chat.id' chat='name' text='test name' onDelete={handleDeleteChat} />
                    <FormComponent onSubmit={onAddChat} />
                </ChatList>
            </Provider>
        );

        expect(component).toMatchSnapshot();
    });
});