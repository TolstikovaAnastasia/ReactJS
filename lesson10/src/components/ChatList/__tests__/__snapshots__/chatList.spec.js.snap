// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatList matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
      >
        <form>
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                id="mui-2"
                placeholder="Type message to send"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-nnbavb"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root  css-jh47zj-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            Send
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </form>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
    >
      <form>
        <div
          class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
              id="mui-2"
              placeholder="Type message to send"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-nnbavb"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root  css-jh47zj-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          Send
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </form>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
